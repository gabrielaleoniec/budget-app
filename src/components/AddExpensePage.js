import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { addExpense } from '../actions/expenses';

const AddExpensePage = (props) => (
    <div>
        <h2>Add expense</h2>
        <ExpenseForm onSubmit={(expense) => {
            props.dispatch(addExpense(expense));
            props.history.push('/');
        }}/>
    </div>
);

const mapStateToProps = (state) => ({
    expenses: state.expenses
})

export default connect(mapStateToProps)(AddExpensePage);