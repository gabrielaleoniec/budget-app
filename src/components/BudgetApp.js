import React from 'react';

const BudgetApp = () => {
    return <div>Inside the compppponent</div>
}

export default BudgetApp;