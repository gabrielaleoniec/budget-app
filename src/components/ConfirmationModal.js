import React from 'react';

const ConfirmationModal = (props) => {
    const className = props.showConfirmationBox ? "confirmation-box--show" : '';
    return <div className="box-layout box-layout--cover">
        <div className={`confirmation-box ${className}`}>
            <p>Should the expense be removed?</p>
            <div className="confirmation-box__buttons">
                <button className="button button--weak" onClick={props.hideConfirmationModal}>Cancel</button>
                <button className="button button--remove" onClick={props.removeExpense}>Remove</button>
            </div>
        </div>
    </div>
}

export default ConfirmationModal;
