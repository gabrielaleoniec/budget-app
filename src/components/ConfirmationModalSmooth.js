import React from 'react';

const ConfirmationModalSmooth = () => {
    return () => {
        
    }
}