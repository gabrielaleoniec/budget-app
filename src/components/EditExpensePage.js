import React from 'react';
import { connect } from 'react-redux';
import moment from 'moment';

import ExpenseForm from './ExpenseForm';
import Confirmation from './ConfirmationModal';
import { startUpdateExpense, startRemoveExpense } from '../actions/expenses';

export class EditExpensePage extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            showConfirmation: false,
            showConfirmationBox: false,
            time: 500
        }
    }
    onSubmit = (expense) => {
        const createdAt = typeof expense.createdAt === 'string' ? moment(expense.createdAt).valueOf() : expense.createdAt;

        this.props.startUpdateExpense(this.props.expense.id, { ...expense, createdAt });
        this.props.history.push('/');
    }

    onRemove = () => {
        this.props.startRemoveExpense({ id: this.props.expense.id });
        this.props.history.push('/');
    }

    showConfirmationModal = () => {
        this.setState({
            showConfirmation: true
        })

        this.openTimeout = setTimeout(() => {
            this.setState({
                showConfirmationBox: true
            })
        }, this.state.time);
    }

    hideConfirmationModal = () => {
        this.setState({
            showConfirmationBox: false
        })

        this.openTimeout = setTimeout(() => {
            this.setState({
                showConfirmation : false
            })
        }, this.state.time);
    }

    addClassname = () => {

    }

    render() {
        if (!this.props.expense || !this.props.expense.id) {
            return <p>Expense for the given id does not exist</p>
        }

        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h2 className="page-header__title">Edit expense</h2>
                    </div>
                </div>
                <div className="content-container">
                    <ExpenseForm onSubmit={this.onSubmit} expense={this.props.expense} />
                    <button onClick={this.showConfirmationModal} className="button button--remove">Remove expense</button>
                    {(this.state.showConfirmation || this.state.showConfirmationBox) && <Confirmation
                        removeExpense={this.onRemove}
                        hideConfirmationModal={this.hideConfirmationModal}
                        time={this.state.time}
                        showConfirmationBox={this.state.showConfirmationBox}
                    />}
                </div>
            </div>
        )
    }
}
const mapDispatchToProps = (dispatch, props) => ({
    startUpdateExpense: (id, expense) => dispatch(startUpdateExpense(id, expense)),
    startRemoveExpense: (data) => dispatch(startRemoveExpense(data))
})

const mapStateToProps = (state, props) => ({
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
})

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);