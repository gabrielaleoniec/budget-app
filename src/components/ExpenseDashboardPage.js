import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilter from './ExpenseListFilter';

const ExpenseDashboardPage = () => (
    <div>
        <h2>Dashboard Page</h2>
        <ExpenseListFilter />
        <ExpenseList />
    </div>
);

export default ExpenseDashboardPage;