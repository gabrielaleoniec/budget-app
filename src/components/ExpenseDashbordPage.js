import React from 'react';

const ExpenseDashbordPage = () => <div>Dashbord Page</div>

export default ExpenseDashbordPage;