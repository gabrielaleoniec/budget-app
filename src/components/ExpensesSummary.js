import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom'
import getExpensesTotal from '../selectors/expenses-total';
import getVisibleExpenses from '../selectors/expenses';

export const ExpensesSummary = ({ totalAmount, expensesCount, allExpensesCount }) => {
    const expenseWord = expensesCount === 1 ? 'expense' : 'expenses';
    const beWord = expensesCount === 1 ? 'is' : 'are';
    const allExpenseWord = allExpensesCount === 1 ? 'expense' : 'expenses';
    const hidenExpensesCount = allExpensesCount - expensesCount;
    return <div className="page-header">
        <div className="content-container">
            <h2 className="page-header__title">There {beWord} <span>{expensesCount}</span> {expenseWord} totalling to <span>{totalAmount}</span></h2>
            {!!hidenExpensesCount && <p>Not showing {hidenExpensesCount} out of {allExpensesCount} {allExpenseWord} because of filters</p>}
            <div>
                <Link to="/create" className="button">Create</Link>
            </div>
        </div>
    </div>
}

const mapStateToProps = (state) => {
    const visibleExpenses = getVisibleExpenses(state.expenses, state.filters);
    return {
        totalAmount: getExpensesTotal(visibleExpenses),
        expensesCount: visibleExpenses.length,
        allExpensesCount: state.expenses.length
    }
}

export default connect(mapStateToProps)(ExpensesSummary);
