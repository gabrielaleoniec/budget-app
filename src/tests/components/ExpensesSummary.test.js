import React from 'react';
import {shallow} from 'enzyme';
import {ExpensesSummary} from '../../components/ExpensesSummary';

test('Should correctly render ExpensesSummary for multiple expenses out of many', () => {
    const wrapper = shallow(<ExpensesSummary totalAmount={1200} expensesCount={3} allExpensesCount={6} />);
    expect(wrapper).toMatchSnapshot();
});

test('Should correctly render ExpensesSummary for single expense out of many', () => {
    const wrapper = shallow(<ExpensesSummary totalAmount={1200} expensesCount={1} allExpensesCount={6} />);
    expect(wrapper).toMatchSnapshot();
});

test('Should correctly render ExpensesSummary for single expense out of one', () => {
    const wrapper = shallow(<ExpensesSummary totalAmount={1200} expensesCount={1} allExpensesCount={1} />);
    expect(wrapper).toMatchSnapshot();
});