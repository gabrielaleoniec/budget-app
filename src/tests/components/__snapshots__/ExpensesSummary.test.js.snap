// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly render ExpensesSummary for multiple expenses out of many 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h2
      className="page-header__title"
    >
      There 
      are
       
      <span>
        3
      </span>
       
      expenses
       totalling to 
      <span>
        1200
      </span>
    </h2>
    <p>
      Not showing 
      3
       out of 
      6
       
      expenses
       because of filters
    </p>
    <div>
      <Link
        className="button"
        to="/create"
      >
        Create
      </Link>
    </div>
  </div>
</div>
`;

exports[`Should correctly render ExpensesSummary for single expense out of many 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h2
      className="page-header__title"
    >
      There 
      is
       
      <span>
        1
      </span>
       
      expense
       totalling to 
      <span>
        1200
      </span>
    </h2>
    <p>
      Not showing 
      5
       out of 
      6
       
      expenses
       because of filters
    </p>
    <div>
      <Link
        className="button"
        to="/create"
      >
        Create
      </Link>
    </div>
  </div>
</div>
`;

exports[`Should correctly render ExpensesSummary for single expense out of one 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h2
      className="page-header__title"
    >
      There 
      is
       
      <span>
        1
      </span>
       
      expense
       totalling to 
      <span>
        1200
      </span>
    </h2>
    <div>
      <Link
        className="button"
        to="/create"
      >
        Create
      </Link>
    </div>
  </div>
</div>
`;
